packages/main/src/themes/Switch.css

```diff
@@ -355,28 +371,4 @@
 .ui5-switch-root .ui5-switch-no-label-icon-off,
 .ui5-switch-root .ui5-switch-icon-off {
 	color: var(--_ui5_switch_text_inactive_color);
-}
-
-/* RTL */
-[dir="rtl"].ui5-switch-root.ui5-switch--checked:not(.ui5-switch--no-label):not(.ui5-switch--semantic) .ui5-switch-slider {
-	transform: var(--_ui5_switch_rtl_transform_with_label);
-}
-
-[dir="rtl"].ui5-switch-root.ui5-switch--checked .ui5-switch-slider {
```

Comment:
seems like this are intended to be here since we add dir attribute to the root element based on the effective dir. So let's keep them as they were. Also as we can see in line: 370 the left and right values are swapped. Seems like a good solution just to do the same for the unhandled case by simply adding:

```css
[dir="rtl"].ui5-switch-root:not(.ui5-switch--checked).ui5-switch--semantic .ui5-switch-text--off,
[dir="rtl"].ui5-switch-root:not(.ui5-switch--checked).ui5-switch-desktop.ui5-switch--no-label .ui5-switch-text--off {
	right: var(--_ui5_switch_text_inactive_left);
	left: var(--_ui5_switch_text_inactive_right);
}
```